tests/gis_tests/geos_tests/test_geos.py

```diff
@@ -1308,6 +1308,20 @@ def test_deconstructible(self):
         self.assertEqual(args, (Point(0, 0), MultiPoint(Point(0, 0), Point(1, 1)), poly))
         self.assertEqual(kwargs, {})
 
+    def test_subclassing(self):
+        class ExtendedPolygon(Polygon):
+            def __init__(self, *args, **kwargs):
+                data = kwargs.pop('data', 0)
+                super(ExtendedPolygon, self).__init__(*args, **kwargs)
+                self._data = data
+
+            def __str__(self):
+                return "EXT_POLYGON - data: %d - %s" % (self._data, self.wkt)
+
+        ext_poly = ExtendedPolygon(((0, 0), (0, 1), (1, 1), (0, 0)), data=3)
+        self.assertEqual(type(ext_poly), ExtendedPolygon)
+        self.assertEqual(str(ext_poly), "EXT_POLYGON - data: 3 - POLYGON ((0 0, 0 1, 1 1, 0 0))")
```

Comment:
Is it unneeded complexity to verify `ext_poly._data` via `__str__()` as opposed to checking the attribute directory?